src/backend/distributed/commands/multi_copy.c

```diff
@@ -465,7 +465,7 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 	 * PG_CATCH. Otherwise, it may be undefined in the PG_CATCH (see sigsetjmp
 	 * documentation).
 	 */
-	shardConnectionHash = CreateShardConnectionHash();
+	copyConnectionHash = CreateShardConnectionHash(MessageContext);
```

Comment:
Previously, no context was passed, but the context flag was set. This results in a brand-new parentless memory context for the hash to use for allocations, i.e. a memory leak.

Simply removing the context flag would be wrong, as that just uses `TopMemoryContext` as the parent instead (not much better).

I figured `MessageContext` would make sense, as this hash only lives for one message (the `COPY`), right?